src/graphql_clj/executor.clj

```diff
@@ -208,35 +210,38 @@
                                    (execute-definition context schema resolver-fn definition fragments variables))
                                  operation-definitions))})))
 
+;; Public API
+
+(defn prepare* [schema-or-state resolver-fn ^String statement]
+  (let [state (if (or (:errors schema-or-state) (:spec-map schema-or-state))
+                schema-or-state
+                (validator/validate-schema schema-or-state))] ;; Schema validation inside execution phase for backwards compatibility
+    (if (:errors state)
+      (select-keys state [:errors])
+      (let [validated-statement (-> statement parser/parse (validator/validate-statement state))]
+        (if (-> validated-statement :state :errors)
+          (select-keys (:state validated-statement) [:errors])
+          (let [resolver (resolver/create-resolver-fn (:schema (:state validated-statement)) resolver-fn)]
+            (assoc-in validated-statement [:state :resolver] resolver)))))))
+
+(def prepare (memoize prepare*))
```

Comment:
It is better not to use memoize. User can choose to use memoize if needed, they can also choose how to memoize. 

Although memoize will get higher performance at the expense of higher memory use. However it is better leave it to user to decide whether use it or not. 

Another good thing about leaving this to the user is that user can choose to use memoize if they wish, or they can use enhanced version of memoize with customized caching policy.